bottlecap/src/logs/aggregator.rs

```diff
@@ -0,0 +1,23 @@
+use crate::logs::processor::IntakeLog;
+
+pub struct Aggregator {
+    messages: Vec<IntakeLog>,
+}
+
+impl Aggregator {
+    pub fn new() -> Self {
+        Aggregator {
+            messages: Vec::new(),
+        }
+    }
+
+    pub fn add(&mut self, log: IntakeLog) {
+        self.messages.push(log);
+    }
+
+    pub fn get_batch(&mut self) -> Vec<IntakeLog> {
+        let messages = self.messages.clone();
```

Comment:
why do you clone messages here?

Reply:
I dediced to just take them so they are empty now